jscomp/gentype/TranslateTypeExprFromTypes.ml

```diff
@@ -201,9 +201,14 @@ let translateConstr ~config ~paramsTranslation ~(path : Path.t) ~typeEnv =
     {dependencies = []; type_ = EmitType.typeReactElement}
   | (["FB"; "option"] | ["option"]), [paramTranslation] ->
     {paramTranslation with type_ = Option paramTranslation.type_}
-  | (["Js"; "Null"; "t"] | ["Js"; "null"]), [paramTranslation] ->
+  | ( (["Js"; "Undefined"; "t"] | ["Undefined"; "t"] | ["Js"; "undefined"]),
```

Comment:
@zth what else from paths hardcoded in this file should be supported for Core, in addition to `Undefined.t` and `Null.t` and `Nullable.t`?

Reply:
Things like `Option.t` or `Int.t` or `Float.t`?
I think we're under the assumption that we're under `"-open RescriptCore"`.

Reply:
`Undefined.t` is getting axed, but the rest of them for sure. Here are a few other candidates (some are perhaps already supported):
- BigInt (https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-2.html#bigint)
- Date
- Map
- Set
- RegExp (?)
- WeakMap
- WeakSet

Reply:
On my phone now can’t verify but I think undefined<…> and null<…> are at the top level of Core for easy usage in code. Don’t need to prefix with a module name if you open Core. Js.Undefined.t is used frequently in my genType bindings - bummed if the Undefined module is being axed.

Reply:
> On my phone now can’t verify but I think undefined<…> and null<…> are at the top level of Core for easy usage in code. Don’t need to prefix with a module name if you open Core. Js.Undefined.t is used frequently in my genType bindings - bummed if the Undefined module is being axed.

Why use `Js.Undefined<t>` when you can use `option<t>`? The definition of `Js.Undefined` predates the runtime representation of `None` as `undefined`.

Reply:
I was writing bindings yesterday with **gentype** and the function I was binding to expected an undefined or T but not a null. I think option, which I used first, compiles to null or undefined or T. I always start with option but sometimes it doesn’t work with gentype.

Reply:
> I was writing bindings yesterday with **gentype** and the function I was binding to expected an undefined or T but not a null. I think option, which I used first, compiles to null or undefined or T. I always start with option but sometimes it doesn’t work with gentype.

Can you give an example of what function you were binding to? Did you get a TypeScript type error, or is it a worry that things don't match?